src/exceptiongroup/_exceptions.py

```diff
@@ -174,12 +217,64 @@ def __repr__(self) -> str:
         return f"{self.__class__.__name__}({self.message!r}, {self._exceptions!r})"
 
 
-class ExceptionGroup(BaseExceptionGroup[E], Exception, Generic[E]):
-    def __new__(cls, __message: str, __exceptions: Sequence[E]) -> ExceptionGroup[E]:
-        instance: ExceptionGroup[E] = super().__new__(cls, __message, __exceptions)
+class ExceptionGroup(BaseExceptionGroup[_ExceptionT_co], Exception):
+    def __new__(cls, __message: str, __exceptions: Sequence[_ExceptionT_co]) -> Self:
+        instance: ExceptionGroup[_ExceptionT_co] = super().__new__(
+            cls, __message, __exceptions
+        )
         if cls is ExceptionGroup:
```

Comment:
Should this be `if issubclass(cls, Exception)` (or `ExceptionGroup`)?  As-is I could subclass `ExceptionGroup` and then store a `BaseException` in that just fine.

Reply:
Yes, this is how the upstream implementation works too. Try running this on Python 3.11:
```
class CustomExcGroup(ExceptionGroup):
    pass

exc = CustomExcGroup("message", [SystemExit(), ValueError()])
```
Works fine. This is somewhat covered in PEP 654: https://peps.python.org/pep-0654/#subclassing-exception-groups

Reply:
Yep, I should probably go ~~complain~~ ask about this upstream 😅